pkg/agent/datamodel/types.go

```diff
@@ -1818,11 +1818,26 @@ type KubeletWebhookAuthorization struct {
 	CacheUnauthorizedTTL Duration `json:"cacheUnauthorizedTTL,omitempty"`
 }
 
-type VMSSInstanceViewCSEStatus struct {
-	// ExitCode stores the exitCode from VMSS CSE output.
+type CSEStatus struct {
+	// ExitCode stores the exitCode from CSE output.
 	ExitCode string `json:"exitCode,omitempty"`
-	// Output stores the output from VMSS CSE output.
+	// Output stores the output from CSE output.
 	Output string `json:"output,omitempty"`
-	// Error stores the error from VMSS CSE output.
+	// Error stores the error from CSE output.
 	Error string `json:"error,omitempty"`
+	// ExecDuration stores the execDuration from CSE output.
+	ExecDuration int `json:"execDuration,omitempty"`
+}
+
+type Error struct {
+	Code    string
+	Message string
+}
+
+func NewError(code, message string) *Error {
+	return &Error{Code: code, Message: message}
+}
+
+func (err *Error) Error() string {
+	return fmt.Sprintf("CSE has invalid message=%s, %s=%s", err.Message, InstanceErrorCode, err.Code)
```

Comment:
%q for the message

Reply:
Also I want to point out that consumders (like RP) should not take this error string format as part of the contract when interpreting the errors, if we do please update it to rely on the new Code property